project11/project11.sql

```diff
@@ -0,0 +1,248 @@
+	UNIQUE(object__id, date),
+	-- Других ключей нет
+    
+    -- Связь N:M между видами соревни объектами. Один вид спорта может проводиться в многих объектах. В одном объекте могут проходить разные виды спорта.
+    FOREIGN KEY(object__id) REFERENCES object_(id)
+);
+
+
+-- Связь inf <-> inf между соревнованиями и спортсменами
+CREATE TABLE SportsmanToCompetition(
+    competition_id INT, 
+    sportsman_id TEXT,
+    -- Между спортсменом и соревнованием существует только одна связь.
+	UNIQUE(competition_id, sportsman_id),
+    -- Других ключей нет
+    
+    -- Каждый спортсмен может участвовать более чем в одном соревновании (но не в одном и том же несколько раз)
+    -- Каждое соревнование может содержать более чем одного спортсмена
+	FOREIGN KEY(competition_id) REFERENCES competition(id),
+	FOREIGN KEY(sportsman_id) REFERENCES sportsman(card_id)
+);
+
+
+-- Тип медали
+CREATE TYPE medal_type AS ENUM('Золото', 'Серебро', 'Бронза');
+
+-- Записи о выигранных медалях
+CREATE TABLE medal(
+    competitions_id INT NOT NULL, 
+    sportsman_id TEXT NOT NULL, 
+    medal MEDAL_TYPE NOT NULL,
+    -- В одном соревновании не может быть выдано две одинаковые медали
+    UNIQUE(competitions_id, medal),
+    -- Каждый спортсмен может получить только одну медаль в одном соревновании.
+	UNIQUE(competitions_id, sportsman_id),
+    -- Других ключей нет
+    
+    -- Связь N:M между видами медалей и соревнованиями. Один вид медали может даваться в разных соревнованиях. В одном соревновании могут быть выданы разные виды медалей
+	FOREIGN KEY(competitions_id) REFERENCES competition(id),
+    -- Связь N:M между видами медалей и спортсменами. Один вид медали может даваться разным спортсменам. Одному спортсмену в разных соревнованиях могут быть выданы разные виды медалей.
+	FOREIGN KEY(sportsman_id) REFERENCES sportsman(card_id)
+);
+
+-- Волонтер определяется своей id картой и при этом волонтеры должны иметь уникальные телефоны.
+-- Может быть неконсистентность, что id волонтера совпадает с id спортсмена.
+CREATE TABLE volunteer(
+    card_id TEXT NOT NULL,
+    name TEXT NOT NULL,
+    -- Один телефон принадлежит только одному волонтеру
+	telephone TEXT NOT NULL UNIQUE,
+    -- card_id может принадлежать только одному волонтеру
+	PRIMARY KEY(card_id)
+    -- Других ключей нет
+);
+
+-- Отношение 1 <-> inf между волонтерами и спортсменами
+
+CREATE TABLE VolunteerToSportsman(
+    volunteer_id TEXT,
+    -- Каждому спортсмену назначается не более одного волонтера.
+    sportsman_id TEXT NOT NULL UNIQUE,
+    -- Других ключей нет
+
+    -- Связь 1:N между волонтерами и спортсменами. Один волонтер может обслуживать много спортсменов.
+	FOREIGN KEY(volunteer_id) REFERENCES volunteer(card_id),
+	FOREIGN KEY(sportsman_id) REFERENCES sportsman(card_id)
+);
+
+-- Отношение inf <-> 1 между заданиями и волонтерами
+-- Задание содержит id волонтера, дату выполнения и текстовое описание.
+CREATE TABLE VolunteerTask(
+    id SERIAL,
+	volunteer_id TEXT NOT NULL,
```

Comment:
Why do you prefer text foreign key over having integer id field in volunteer table and using it as a foreign key?

Reply:
done